### ISSUE TITLE
`self.app.get(url)` always includes an explicit port in `HTTP_HOST` header, even for default ports

### ISSUE THREAD

**Issue by mrmachine** [django-webtest/django-webtest#124]
We have some redirect tests which make a `self.app.get(url)` request and then check the response is a redirect to an expected location.

Our URL going in does not include a port. E.g. `http://subdomain.example.com/foo/bar/` and we check that the response location is the same with subdomain stripped or replaced.

The request object created by `self.app.get()` will include `HTTP_HOST` header set to `subdomain.example.com:80`, which then causes our test to fail.

Now, we can work around this with our test by always passing in a URL with an explicitly defined port, or stripping the port from the returned location, but this is a surprising difference between a regular browser request and the request created by `self.app.get()`.

```
In [1]: from django_webtest import TestApp

In [2]: from django.core.wsgi import get_wsgi_application

In [3]: application = get_wsgi_application()

In [4]: testapp = TestApp(application)

In [5]: response = testapp.get('http://example.com')

In [6]: response.request.headers['HOST']
Out[6]: 'example.com:80'
```

If you look at a regular django request from a browser, to a default port (80, 443), the `HTTP_HOST` header will not include the port number.

**Comment by gawel** [comment]
Looks like the "problem" is in webob https://github.com/Pylons/webob/blob/master/src/webob/request.py#L1515

**Comment by mrmachine** [comment]
Indeed. And it looks like not in a place where we could easily override it in a subclass.

Also looks like we already work around a similar issue for `response.location`: https://github.com/django-webtest/django-webtest/blob/6a7bab7ffca6b829c870a303617443263a327517/django_webtest/response.py#L44-L45

Is django or webob more correct in adding default ports to the `HOST` header? webob might not be interested in changing their behaviour to match django's.

**Comment by gawel** [comment]
I don't know. What I do know is that webtest is used to test wsgi applications. Not external urls.

Means that you're no supposed to call get/post/etc. with full urls. Only the path portion of the url.

If testserver is not ok for you, then pass  your HTTP_HOST to extra_environ. See https://github.com/django-webtest/django-webtest/issues/102
